addon/mixins/form-data-adapter.js

```diff
@@ -28,20 +28,25 @@ export default Ember.Mixin.create({
     var root = Object.keys(data)[0];
 
     Object.keys(data[root]).forEach(function(key) {
-      if (typeof data[root][key] !== 'undefined') {
-        var value = data[root][key] === null ? '' : data[root][key];
-        var formKey = this.get('disableRoot') ? key : root + "[" + key + "]";
-
-        if (Ember.isArray(value)) {
-          value.forEach(function(item) {
-            formData.append(formKey + '[]', item);
-          }, this);
-        } else {
-          formData.append(formKey, value);
-        }
-      }
+      var baseFormKey = this.get('disableRoot') ? key : `${root}[${key}]`;
+
+      this._appendValue(data[root][key], baseFormKey, formData);
     }, this);
 
     return formData;
-  }
+  },
+
+  _appendValue(value, formKey, formData) {
+    if (Ember.isArray(value)) {
+      value.forEach(function(item) {
+        this._appendValue(item, `${formKey}[]`, formData);
+      }, this);
+    } else if (typeof value === 'object' && value !== null) {
```

Comment:
@GCorbel that's actually correct - and seems like a blunder on my side. I used leaned-down version of modifications proposed in this PR in the project I'm currently working on, apparently this check is something I missed when moving changes over. Will check this twice and add test case to be sure - I'm more and more leaning towards handling just plain objects here (`Object` instances).